pkg/logs/launchers/integration/launcher_test.go

```diff
@@ -378,24 +378,20 @@ func (suite *LauncherTestSuite) TestCreateFileAfterScanInitialFile() {
 func (suite *LauncherTestSuite) TestSentLogExceedsTotalUsage() {
 	suite.s.combinedUsageMax = 3 * 1024 * 1024
 
+	// Given 3 files exist
 	filename1 := "sample_integration1_123.log"
 	filename2 := "sample_integration2_123.log"
 	filename3 := "sample_integration3_123.log"
+	files := [3]string{filename1, filename2, filename3}
 
-	file1, err := os.Create(filepath.Join(suite.s.runPath, filename1))
-	assert.Nil(suite.T(), err)
-	file2, err := os.Create(filepath.Join(suite.s.runPath, filename2))
-	assert.Nil(suite.T(), err)
-	file3, err := os.Create(filepath.Join(suite.s.runPath, filename3))
-	assert.Nil(suite.T(), err)
-
+	//  And I write 1Mb to each file in seq order
 	dataOneMB := make([]byte, 1*1024*1024)
-	file1.Write(dataOneMB)
-	file2.Write(dataOneMB)
-	file3.Write(dataOneMB)
-	file1.Close()
-	file2.Close()
-	file3.Close()
+	for _, filename := range files {
+		file, err := os.Create(filepath.Join(suite.s.runPath, filename))
+		require.NoError(suite.T(), err)
+		_, _ = file.Write(dataOneMB)
+		_ = file.Close()
+	}
```

Comment:
Lacking any guaranteed precision from both the system clock and the filesystem timestamps I think this is still rather fragile. It is rather likely that all three files accessed so close together will have the same modification timestamp. Not hard to get this effect even with a shell:
```
for i in foo bar baz; do echo $i > $i; done; stat foo bar baz | grep Modify
Modify: 2024-10-07 14:16:01.084963899 +0000
Modify: 2024-10-07 14:16:01.084963899 +0000
Modify: 2024-10-07 14:16:01.084963899 +0000
```

Reply:
Yeah, that was my concern. I was debating adding a sleep of a few `ms` after the `file.close()`. I'm hoping as we create the file in the loop before writing that it should cause enough time to stop the files having the same timestamp. Quickly running a similar function as what we see in the tests (see playground [here](https://go.dev/play/p/i5SofD4qe8V)) I get something like:

```
file_1 Modify time: 2024-10-07 14:57:35.115021841 +0000 UTC
file_2 Modify time: 2024-10-07 14:57:35.116025571 +0000 UTC
file_3 Modify time: 2024-10-07 14:57:35.116973051 +0000 UTC
```